x/emissions/keeper/msgserver/msg_server_reputer_payload.go

```diff
@@ -0,0 +1,72 @@
+package msgserver
+
+import (
+	"context"
+	"github.com/allora-network/allora-chain/x/emissions/types"
+	mintTypes "github.com/allora-network/allora-chain/x/mint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// A tx function that accepts a individual loss and possibly returns an error
+func (ms msgServer) InsertReputerPayload(ctx context.Context, msg *types.MsgInsertReputerPayload) (*types.MsgInsertReputerPayloadResponse, error) {
+	err := checkInputLength(ctx, ms, msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the topic exists
+	topicExists, err := ms.k.TopicExists(ctx, msg.TopicId)
+	if err != nil || !topicExists {
+		return nil, types.ErrInvalidTopicId
+	}
+
```

Comment:
code is not correct since we are inserting the reputer nonce. 

When the reputer nonce gets inserted, the situation needs to be: 
* there IS NOT an unfulfilled **worker** nonce for that worker (ie the nonce has been fulfilled)
* there IS an unfulfilled **reputer** nonce

Reply:
yes, you're correct